.github/workflows/ui-run-tests.yml

```diff
@@ -0,0 +1,37 @@
+name: '[UI] Run tests'
+
+# Automatically trigger for changes to ./ui/* on push/PR
+on:
+  push:
+    paths:
+      - 'ui/**'
+    branches-ignore:
+      - main
+
+  pull_request:
+    paths:
+      - 'ui/**'
+    branches-ignore:
+      - main
+
+  workflow_dispatch:
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+
+    # Set working directory for entire test
+    defaults:
+      run:
+        working-directory: ui
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Build with Gradle
+        run: npm test
```

Comment:
I'd also like to run `npm run typecheck` and `npm run lint:report`, since our automated verification is split into multiple passes in the JS world

Reply:
I was planning to add that as a separate check, but fine either way 🤷